qiskit/circuit/library/standard_gates/pauli.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Simulator command to perform multiple pauli gates in a single pass
+"""
+import numpy as np
+
+from qiskit.circuit.quantumregister import QuantumRegister
+from qiskit.circuit.library.standard_gates.i import IGate
+from qiskit.circuit.library.standard_gates.x import XGate
+from qiskit.circuit.library.standard_gates.y import YGate
+from qiskit.circuit.library.standard_gates.z import ZGate
+
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.exceptions import CircuitError
+
+
+class PauliGate(Gate):
+    r"""A multi-qubit Pauli gate.
+
+        This gate exists for optimization purposes for the
+        quantum statevector simulation, since applying multiple
+        pauli gates to different qubits at once can be done via
+        a single pass on the statevector.
+
+        The functionality is equivalent to applying
+        the pauli gates sequentially using standard Qiskit gates
+        """
+
+    def __init__(self, pauli_string="I"):
```

Comment:
I would change `pauli_string` to `label` to match the label parameter of the `Pauli` and `Operator` classes

```suggestion
    def __init__(self, label="I"):
```

Reply:
I would actually remove the default kwarg for this and make it require putting in a string. It doesn't make much sense to add an identity Pauli to a circuit through `PauliGate()`

Reply:
I completely agree; it is there only because tests fail without it, and I did not want to change the testing framework too much in this PR.

Reply:
Got rid of these and fixed the tests.